### PR TITLE
[INLONG-1712][Feature][Agent] Add agent Prometheus metric statistics

### PR DESCRIPTION
**Pull request by shink**
Fixes: #1712

### Motivation

Expose agent metrics by prometheus. See #1712.

### Modifications

- inlong-agent/agent-core
- inlong-agent/agent-common

### CODE REVIEW

File: inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/constants/AgentConstants.java [hunk 1]
```diff
@@ -160,4 +160,7 @@
 
     public static final String AGENT_LOCAL_IP = "agent.local.ip";
 
+    public static final String PROMETHEUS_EXPORTER_PORT = "agent.prometheus.exporter.port";
+    public static final int DEFAULT_PROMETHEUS_EXPORTER_PORT = 8080;
```
Comment (healchow) [inline inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/constants/AgentConstants.java:164]: It is recommended to use some infrequently used ports.

File: inlong-agent/conf/agent.properties [hunk 2]
```diff
@@ -101,3 +101,10 @@ channel.memory.capacity=5000
 agent.manager.vip.http.host=
 agent.manager.vip.http.port=
 
+
+
+############################
+# prometheus config
+############################
+# prometheus exporter server default port
+agent.prometheus.exporter.port=8080
```
Comment (healchow) [inline inlong-agent/conf/agent.properties:110]: It is recommended to use some infrequently used ports.

File: inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/constants/AgentConstants.java (hunk 1 above)
Comment (dockerzhang) [reply inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/constants/AgentConstants.java:164]: @healchow the default port for Prometheus is 8080

File: inlong-agent/conf/agent.properties (hunk 2 above)
Comment (dockerzhang) [reply inlong-agent/conf/agent.properties:110]: @shink it's better to add a config to enable or disable expose Prometheus HTTP server.

**Comment by EMsnap** [comment]
Pls add some uts

**Comment by gosonzhang** [comment]
Are you still paying attention to this issue?

Someone has suggested some amendments, please confirm them, thanks!

File: inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/constants/AgentConstants.java [hunk 3]
```diff
@@ -160,4 +160,10 @@
 
     public static final String AGENT_LOCAL_IP = "agent.local.ip";
 
+    public static final String PROMETHEUS_ENABLE = "agent.prometheus.enable";
+    public static final boolean DEFAULT_PROMETHEUS_ENABLE = false;
+
```
Comment (dockerzhang) [inline inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/constants/AgentConstants.java:165]: what's difference about `PROMETHEUS_ENABLE` and  `DEFAULT_PROMETHEUS_ENABLE`?

**Comment by dockerzhang** [comment]
the agent has exposed many JMX metrics, it's better to reuse them.
https://inlong.apache.org/docs/next/modules/agent/overview#monitor-metrics-configuration

**Comment by shink** [comment]
@dockerzhang @gosonzhang @healchow Please review.

**Review by dockerzhang — APPROVED** [review]
LGTM

**Review by healchow — APPROVED** [review]
+1

**Review by gosonzhang — APPROVED** [review]
+1

**Pull request merged**